docs/dev/table/udfs.md

```diff
@@ -112,50 +152,54 @@ public class HashCode extends ScalarFunction {
 }
 '''
 
+class HashCode(ScalarFunction):
+  def eval(self, s):
+    return hash(s) * factor
+
 table_env = BatchTableEnvironment.create(env)
 
-# register the java function
+# register the Java function
 table_env.register_java_function("hashCode", "my.java.function.HashCode")
 
+# register the Python function
+table_env.register_function("py_hash_code", udf(HashCode(), DataTypes.STRING(), DataTypes.BIGINT()))
+
 # use the function in Python Table API
-my_table.select("string, string.hashCode(), hashCode(string)")
+my_table.select("string, string.hashCode(), hashCode(string), string.py_hash_code(), py_hash_code(string)")
```

Comment:
String type has not been supported by now for the python udfs. We can use bigint for now and change to string later.